### ISSUE TITLE
ERROR: BoundsError: attempt to access 0-element Array{SharedArrays.SharedArray{Float64,2},1} at index [1]

### ISSUE THREAD

**Issue by mattwilliamson13** [Circuitscape/Circuitscape.jl#201]
Hello, 
I am consistently getting the following error:
`ERROR: BoundsError: attempt to access 0-element Array{SharedArrays.SharedArray{Float64,2},1} at index [1]`

when trying to run in pairwise mode for a relatively large problem (statewide at 270m resolution). This error does not show up in all-to-one mode nor in any of the tests, but consistently shows up in pairwise. For context, I am trying to run a "wall to wall" analysis and so all of the nodes are on the edge of the landscape. I don't think this should matter, but I can't seem to figure out why this is failing and the error doesn't provide me many clues. The trace is:

```
Stacktrace:
 [1] (::getfield(Base, Symbol("##699#701")))(::Task) at ./asyncmap.jl:178
 [2] foreach(::getfield(Base, Symbol("##699#701")), ::Array{Any,1}) at ./abstractarray.jl:1920
 [3] maptwice(::Function, ::Channel{Any}, ::Array{Any,1}, ::UnitRange{Int64}) at ./asyncmap.jl:178
 [4] wrap_n_exec_twice at ./asyncmap.jl:154 [inlined]
 [5] #async_usemap#684 at ./asyncmap.jl:103 [inlined]
 [6] #async_usemap at ./none:0 [inlined]
 [7] #asyncmap#683 at ./asyncmap.jl:81 [inlined]
 [8] #asyncmap at ./none:0 [inlined]
 [9] #pmap#213(::Bool, ::Int64, ::Nothing, ::Array{Any,1}, ::Nothing, ::typeof(Distributed.pmap), ::Function, ::Distributed.WorkerPool, ::UnitRange{Int64}) at /Users/sabae/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.2/Distributed/src/pmap.jl:126
 [10] pmap(::Function, ::Distributed.WorkerPool, ::UnitRange{Int64}) at /Users/sabae/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.2/Distributed/src/pmap.jl:101
 [11] #pmap#223(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Distributed.pmap), ::Function, ::UnitRange{Int64}) at /Users/sabae/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.2/Distributed/src/pmap.jl:156
 [12] pmap at /Users/sabae/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.2/Distributed/src/pmap.jl:156 [inlined]
 [13] amg_solver_path(::Circuitscape.GraphData{Float64,Int32}, ::Circuitscape.RasterFlags, ::Dict{String,String}, ::Bool) at /Users/mattwilliamson/.julia/packages/Circuitscape/mKAhh/src/core.jl:221
 [14] single_ground_all_pairs(::Circuitscape.GraphData{Float64,Int32}, ::Circuitscape.RasterFlags, ::Dict{String,String}, ::Bool) at /Users/mattwilliamson/.julia/packages/Circuitscape/mKAhh/src/core.jl:55
 [15] _pt_file_polygons_path(::Circuitscape.RasData{Float64,Int32}, ::Circuitscape.RasterFlags, ::Dict{String,String}) at /Users/mattwilliamson/.julia/packages/Circuitscape/mKAhh/src/raster/pairwise.jl:105
 [16] raster_pairwise(::Type, ::Type, ::Dict{String,String}) at /Users/mattwilliamson/.julia/packages/Circuitscape/mKAhh/src/raster/pairwise.jl:27
 [17] _compute(::Type, ::Type, ::Dict{String,String}) at /Users/mattwilliamson/.julia/packages/Circuitscape/mKAhh/src/run.jl:42
 [18] macro expansion at ./util.jl:213 [inlined]
 [19] compute(::String) at /Users/mattwilliamson/.julia/packages/Circuitscape/mKAhh/src/run.jl:31
 [20] top-level scope at REPL[7]:1
```

**Comment by mattwilliamson13** [comment]
I uploaded the files [here](https://drive.google.com/file/d/118k1vNWHNkOR9jMfO2UBnwiLh3tIpdbj/view?usp=sharing) in case you want to try reproducing the error

**Comment by ranjanan** [comment]
Thanks for reporting this! I'm able to reproduce this on my machine so let me investigate

**Comment by ranjanan** [comment]
@mattwilliamson13 I found the bug and merged the fix via #207. Could you run `Pkg.add(PackageSpec(name="Circuitscape", rev = "master"))` at the Julia prompt? That would upgrade you to the master branch and you can check if the fix works.

**Comment by mattwilliamson13** [comment]
Awesome. Thanks @ranjanan. Works great!
